src/intent_parser/lookup_intent_parser.rs

```diff
@@ -0,0 +1,824 @@
+    shared_resources: Arc<SharedResources>,
+}
+
+impl LookupIntentParser {
+    /// load parser from the file system
+    pub fn from_path<P: AsRef<Path>>(
+        path: P,
+        shared_resources: Arc<SharedResources>,
+    ) -> Result<Self> {
+        let parser_model_path = path.as_ref().join("intent_parser.json");
+        let model_file = File::open(&parser_model_path).with_context(|_| {
+            format!(
+                "Cannot open LookupIntentParser file '{:?}'",
+                &parser_model_path
+            )
+        })?;
+        let model: LookupParserModel = serde_json::from_reader(model_file)
+            .with_context(|_| "Cannot deserialize LookupIntentParser json data")?;
+        Self::new(model, shared_resources)
+    }
+}
+
+impl LookupIntentParser {
+    /// create a parser instance
+    pub fn new(model: LookupParserModel, shared_resources: Arc<SharedResources>) -> Result<Self> {
+        let language = Language::from_str(&model.language_code)?;
+        Ok(LookupIntentParser {
+            language,
+            slots_names: model.slots_names,
+            intents_names: model.intents_names,
+            map: model.map,
+            ignore_stop_words: model.config.ignore_stop_words,
+            shared_resources,
+        })
+    }
+}
+
+impl LookupIntentParser {
+    fn generate_key(&self, input: &str) -> String {
+        let lang = NluUtilsLanguage::from_language(self.language);
+        let empty_set = HashSet::new();
+        let stop_words = if self.ignore_stop_words {
+            &self.shared_resources.stop_words
+        } else {
+            &empty_set
+        };
+        let key = tokenize_light(input, lang)
+            .iter()
+            .filter(|tkn| !stop_words.contains(tkn.as_str()))
+            .fold(String::new(), |mut acc, tkn| {
+                acc.push_str(tkn);
+                acc
+            });
+        key.to_lowercase()
+    }
+
+    fn get_all_entities(&self, input: &str) -> Result<Vec<MatchedEntity>> {
+        // get builtin entities
+        let b = self
+            .shared_resources
+            .builtin_entity_parser
+            .extract_entities(input, None, true)?
+            .into_iter()
+            .map(|entity| entity.into());
+        // get custom entities
+        let c = self
+            .shared_resources
+            .custom_entity_parser
+            .extract_entities(input, None)?
+            .into_iter()
+            .map(|entity| entity.into());
+
+        // combine entities
+        let mut all_entities: Vec<MatchedEntity> = Vec::new();
+        all_entities.extend(b);
+        all_entities.extend(c);
+
+        Ok(deduplicate_overlapping_entities(all_entities))
+    }
+
+    fn empty_parsing_result(&self) -> InternalParsingResult {
+        InternalParsingResult {
+            intent: IntentClassifierResult {
+                intent_name: None,
+                probability: 1.0,
+            },
+            slots: vec![],
+        }
+    }
```

Comment:
This should probably be a method of `InternalParsingResult` (`InternalParsingResult::empty()`), and could be used by the `DeterministicIntentParser` as well.